sklearn/base.py

```diff
@@ -920,7 +920,7 @@ def get_feature_names_out(self, input_features=None):
         feature_names_out : ndarray of str objects
             Transformed feature names.
         """
-        check_is_fitted(self, "_n_features_out")
+        check_is_fitted(self, ["_n_features_out", "n_features_in_"])
```

Comment:
I can see this is failing `check_is_fitted` is not working with the `@property` in `BaseRandomProjection`. May you revert this change and apply this patch, such that `_n_features_out` is defined and works with `check_is_fitted`?

```patch
diff --git a/sklearn/random_projection.py b/sklearn/random_projection.py
index c780de1e95..9a11e6e133 100644
--- a/sklearn/random_projection.py
+++ b/sklearn/random_projection.py
@@ -419,16 +419,10 @@ class BaseRandomProjection(
         if self.compute_inverse_components:
             self.inverse_components_ = self._compute_inverse_components()
 
+        # Required by ClassNamePrefixFeaturesOutMixin.get_feature_names_out.
+        self._n_features_out = self.n_components
         return self
 
-    @property
-    def _n_features_out(self):
-        """Number of transformed output features.
-
-        Used by ClassNamePrefixFeaturesOutMixin.get_feature_names_out.
-        """
-        return self.n_components
-
     def inverse_transform(self, X):
         """Project data back to its original space.
```